conda/misc.py

```diff
@@ -259,7 +259,7 @@ def clone_env(prefix1, prefix2, verbose=True, quiet=False, index_args=None):
         what = "Package%s " % ('' if len(notfound) == 1 else 's')
         notfound = '\n'.join(' - ' + fn for fn in notfound)
         msg = '%s missing in current %s channels:%s' % (what, context.subdir, notfound)
-        raise CondaRuntimeError(msg)
+        raise PackageNotFoundError(msg)
```

Comment:
We still need to clean up this error now.  Most all of this message construction should be done inside the exception class itself.  Looking at other uses of this error, you'll need to use `location `as an input argument to the exception, where location is either usually a prefix or list of channels it looks like.